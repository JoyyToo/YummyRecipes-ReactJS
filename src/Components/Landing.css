.App {
    text-align: center;
}
p{
  font-family: 'Josefin Slab', serif;
}

.intro-message>h1 {
  margin-top: 200px;
  margin-bottom: 0px;
  text-shadow: 2px 2px 3px rgba(0, 0, 0, 0.6);
  font-size: 5em;
}

.intro-divider {
  width: 400px;
  margin-bottom: 50px;
  border-top: 1px solid #f8f8f8;
  border-bottom: 1px solid rgba(0, 0, 0, 0.2);
}

.intro-message>h2 {
  text-shadow: 2px 2px 3px rgba(0, 0, 0, 0.6);
}

@media(max-width:767px) {
    .intro-message {
      padding-bottom: 15%;
    }
    .intro-message>h1 {
      font-size: 3em;
    }
    .intro-social-buttons{
      font-size: 80%;
      display: block;
      margin-bottom: 20px;
      padding: 0;
    }
    .intro-divider {
      width: 100%;
    }
  }
  
.network-name {
    border-radius: 25px;
    text-transform: uppercase;
    font-size: 14px;
    font-weight: bold;
    letter-spacing: 2px;
    border-width: 2px;
    background: rgba(20, 20, 20, 0);
    color: white;
    height: 50px;
    margin: 13px 0px 12px 10px;
    margin-top: 10px;
    width: 130px;
    margin-bottom: 220px;
  }

.network-name:hover {
    background-color: white;
    transition-duration: 0.4s;
    color: black;
}
.navbar-brand{
  color: white;
  background-color: rgba(20, 20, 20, 0);
  opacity: .8;
  align-content: right !important; 
  text-decoration: none;
}

.navbar-brand h2 {
  color: white;
  margin-left: 20px;
}

#container {
  background-color: rgba(20, 20, 20, 0);
  margin-left: 40px; 
}