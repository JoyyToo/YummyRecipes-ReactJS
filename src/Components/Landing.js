import React from 'react';
import { Link } from 'react-router-dom';
import './Landing.css'

const landing = () => { 
    const Accesstoken = window.localStorage.token

    return  (       
    <div className="App">
        <div className="intro-message">
            <h1 >Yummy recipes</h1> <br />
            <h2>Makes your cooking easier and better</h2>
            <hr className="intro-divider" />
        </div>
        <div className="list-inline intro-social-buttons">
            <div className="list-inline-item">
                { !Accesstoken ? (
                    <div>
                        <Link to="/login">
                        <button className="network-name">LOG IN</button></Link>
                        <Link to="/register">
                        <button className="network-name">SIGN UP</button></Link>
                    </div>
                    ) : (
                        <div>
                        <Link to="/addcategory">
                        <button className="network-name">ADD CATEGORIES</button></Link>
                    
                        <Link to="/categories">
                        <button className="network-name">VIEW CATEGORIES</button></Link>
                        </div>
                )}
  
            </div>
        </div>
        <div className="copyright">
            <p>Copyright &copy; Yummy Recipes 2018. All Rights Reserved</p>
        </div>

    </div>
    )
};

export default landing;

