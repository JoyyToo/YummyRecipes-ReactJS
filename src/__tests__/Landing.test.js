import React from 'react';
import { shallow, mount } from 'enzyme';
import toJson, { shallowToJson } from 'enzyme-to-json';

import Landing from '../Components/Landing';


describe('Landing component', () => {
  const wrapper = shallow(<Landing />);

  it('renders properly', () => {
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });

  it('renders component in divs', () => {
    expect(wrapper.find('div')).toHaveLength(6);
  });

  it('renders component in h2', () => {
    expect(wrapper.find('h2')).toHaveLength(1);
  });

  it('renders component in h1', () => {
    expect(wrapper.find('h1')).toHaveLength(1);
  });

  it('renders component in hr', () => {
    expect(wrapper.find('hr')).toHaveLength(1);
  });

  it('renders component in button', () => {
    expect(wrapper.find('button')).toHaveLength(2);
  });

  it('renders component in p', () => {
    expect(wrapper.find('p')).toHaveLength(1);
  });

})