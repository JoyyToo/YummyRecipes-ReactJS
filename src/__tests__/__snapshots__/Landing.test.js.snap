// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Landing component renders properly 1`] = `
<div
  className="App"
>
  <div
    className="intro-message"
  >
    <h1>
      Yummy recipes
    </h1>
     
    <br />
    <h2>
      Makes your cooking easier and better
    </h2>
    <hr
      className="intro-divider"
    />
  </div>
  <div
    className="list-inline intro-social-buttons"
  >
    <div
      className="list-inline-item"
    >
      <div>
        <Link
          replace={false}
          to="/login"
        >
          <button
            className="network-name"
          >
            LOG IN
          </button>
        </Link>
        <Link
          replace={false}
          to="/register"
        >
          <button
            className="network-name"
          >
            SIGN UP
          </button>
        </Link>
      </div>
    </div>
  </div>
  <div
    className="copyright"
  >
    <p>
      Copyright Â© Yummy Recipes 2018. All Rights Reserved
    </p>
  </div>
</div>
`;
